src/main/kotlin/no/nav/syfo/aktivitetskrav/domain/AktivitetskravVurdering.kt

```diff
@@ -1,24 +1,99 @@
 package no.nav.syfo.aktivitetskrav.domain
 
 import no.nav.syfo.aktivitetskrav.api.AktivitetskravVurderingResponseDTO
+import no.nav.syfo.aktivitetskrav.api.Arsak
 import no.nav.syfo.aktivitetskrav.database.PAktivitetskravVurdering
 import no.nav.syfo.util.nowUTC
 import java.time.LocalDate
 import java.time.OffsetDateTime
 import java.util.*
 
-enum class VurderingArsak(val validForStatus: AktivitetskravStatus) {
-    OPPFOLGINGSPLAN_ARBEIDSGIVER(AktivitetskravStatus.AVVENT),
-    INFORMASJON_BEHANDLER(AktivitetskravStatus.AVVENT),
-    DROFTES_MED_ROL(AktivitetskravStatus.AVVENT),
-    DROFTES_INTERNT(AktivitetskravStatus.AVVENT),
-    ANNET(AktivitetskravStatus.AVVENT),
-    MEDISINSKE_GRUNNER(AktivitetskravStatus.UNNTAK),
-    TILRETTELEGGING_IKKE_MULIG(AktivitetskravStatus.UNNTAK),
-    SJOMENN_UTENRIKS(AktivitetskravStatus.UNNTAK),
-    FRISKMELDT(AktivitetskravStatus.OPPFYLT),
-    GRADERT(AktivitetskravStatus.OPPFYLT),
-    TILTAK(AktivitetskravStatus.OPPFYLT);
+sealed class VurderingArsak {
```

Comment:
Tenker egentlig det er litt dumt at domene modellen her må ha `toString` og `fromString` metoder ettersom den da implisitt har et forhold til lagring og henting 🤔

Reply:
Går det ikke an å ha streng-verdiene som en value for hver enum i Arsak?

Reply:
Hadde det gått an å flyttet fromString-greiene til P-objektet og toString til Dto-en da?

Reply:
> Går det ikke an å ha streng-verdiene som en value for hver enum i Arsak?

Dette er vel når det er snakk om `enum` og ikke `sealed class`? Tror ikke helt jeg skjønner hva du mener her 😊

> Hadde det gått an å flyttet fromString-greiene til P-objektet og toString til Dto-en da?

Trenger `fromString` både når responsen kommer inn via API'et, og når man henter fra DB 🤔 Forsøker å legge på en funksjon på `Arsak` enumen 😊

Reply:
Foreslår å flytte Arsak-enumen med `toVurderingArsak` til egen Arsak-fil i domain (siden den brukes både fra api og db).

Reply:
Lurer på om jeg har fått til en bedre struktur nå. Ble på en måte litt forslaget ditt Anders. Det er jo ikke noe i veien for at domenet har en spesifikk verdi som betyr de forskjellige årsakene, så fikset litt på `VurderingArsak` klassestrukturen.